core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -537,6 +538,7 @@ class KafkaController(val config: KafkaConfig,
 
   private def removeMetrics(): Unit = {
     KafkaController.MetricNames.foreach(metricsGroup.removeMetric)
+    controllerContext.stats.removeMetrics()
```

Comment:
we should be registering these metrics in initializeControllerContext and removing them in resetControllerContext. The responsibility of correctly resetting and initializing controllerContext is that of KafkaController.

Reply:
My thought is: this PR is to remove the ControllerStats metric when the broker shutdown. Before that, every broker in the cluster has registered all the metrics of ControllerStats. If following your suggestion, then only the real controller broker will register the metrics of ControllerStats, while other brokers do not. I don’t know this Is a logical change of a piece necessary? @divijvaidya

Reply:
I first put the registration and removal of the metric of `ControllerStats` into `initializeControllerContext` and `resetContext`.
In addition, I also called `resetContext` in all the places where the `ControllerContext` was instantiated in the unit tests, so as to remove the `ControllerStats` metric.